src/App.js

```diff
@@ -10,17 +10,94 @@ class App extends Component {
 
     this.state = {
       smoothies: smoothies,
-      ingredients: ingredients
+      ingredients: ingredients,
+      error: '',
+      chosenIngredients: [],
+      // filteredSmoothies: smoothies
     }
   }
 
+  fetchData = () => {
+    fetch('http://whateverly-datasets.herokuapp.com/api/v1/smoothies')
+    .then(response => response.json())
+    .then(result => {
+      this.setState({
+        smoothies: result.smoothies
+      })
+    })
+    .catch(error => {
+      this.setState({error: error.message})
+    })
+    fetch('http://whateverly-datasets.herokuapp.com/api/v1/ingredients')
+      .then(response => response.json())
+      .then(result => {
+        this.setState({
+          ingredients: result.ingredients
+        })
+      })
+      .catch(error => {
+      this.setState({error: error.message})
+      })
+  }
+  
+  componentDidMount() {
+    this.fetchData()
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.chosenIngredients.length !== this.state.chosenIngredients.length) {
+      this.filterSmoothies()
+    }
+  }
+
+  chooseIngredients = (ingredient) => {
+    const updatedIngredients = [ ...this.state.chosenIngredients, ingredient]
+
+    this.setState({ chosenIngredients: updatedIngredients })
+  }
+
+  removeIngredient = (ingredient) => {
+    const updatedIngredients = this.state.chosenIngredients.filter(chosenIngredient => {
+      return chosenIngredient !== ingredient
+    })
+
+    this.setState({ chosenIngredients: updatedIngredients })
+  }
+
+  filterSmoothies = () => {
+    const { chosenIngredients } = this.state
+    // keep an updated list of smoothies that follow the rules
+
+    const matching = this.state.smoothies.filter(smoothie => {
+      //array of all things that match
+      let matched = false
+
+       chosenIngredients.forEach(chosen=>{
+          matched = smoothie.ingredients.includes(chosen)
+       })
+
+       console.log(matched)
```

Comment:
Like the logic in this, but let's remove the console logs and clean up the function a bit.

Reply:
I agree ⬆️